core/src/services/hdfs/backend.rs

```diff
@@ -262,12 +317,36 @@ impl Accessor for HdfsBackend {
             open_options.write(true);
         }
 
-        let f = open_options
-            .async_open(&p)
-            .await
-            .map_err(new_std_io_error)?;
+        if let Some(tmp_path) = &tmp_path {
+            let mut t = open_options
+                .async_open(tmp_path.to_str().unwrap())
+                .await
+                .map_err(new_std_io_error)?;
 
-        Ok((RpWrite::new(), HdfsWriter::new(f)))
+            let tmp_path = tmp_path.clone();
+            let target_path = target_path.clone();
+
+            t.flush().await.map_err(new_std_io_error)?;
+            t.close().await.map_err(new_std_io_error)?;
+
+            self.client
+                .rename_file(tmp_path.to_str().unwrap(), target_path.to_str().unwrap())
```

Comment:
I didn't get this. Why we can't finish this in `HdfsWriter`?

Reply:
This is exactly what I wanted to do at the beginning.So can we initialize `client: Arc<hdrs::Client>`?
```rust
pub struct HdfsWriter<F> {
    f: F,
    client: Arc<hdrs::Client>,
}

impl<F> HdfsWriter<F> {
    pub fn new(f: F,client: Arc<hdrs::Client>) -> Self {
        Self {
            f,
            client,
        }
    }
}


#[async_trait]
impl oio::Write for HdfsWriter<hdrs::AsyncFile> {
    fn poll_close(&mut self, cx: &mut Context<'_>) -> Poll<Result<()>> {
            let tmp_path = self.tmp_path.clone();
            let target_path = self.target_path.clone();
            
            Pin::new(&mut self.f)
            .poll_close(cx)
            .map_err(new_std_io_error)
            
              self.client
                .rename_file(tmp_path.to_str().unwrap(), target_path.to_str().unwrap())
                .map_err(new_std_io_error)?;
    }
}
```
This requires us to introduce into `self.client`
```rust
Ok((RpWrite::new(), HdfsWriter::new(f,target_path,tmp_path,Self.client)))
```

Reply:
It's ok to passing an `Arc<hdrs::Client>` into the writer.

Reply:
It reproduces the error of cluster CI.I'll fix it later.And now I am leaving home.
```shell
failures:
    behavior::test_fuzz_appender
    behavior::test_append_create_append
    behavior::test_appender_futures_copy

test result: FAILED. 102 passed; 3 failed; 0 ignored; 0 measured; 0 filtered out; finished in 13.31s

```